src/components/flowchart/components/sliced-pipeline-action-bar/sliced-pipeline-action-bar.js

```diff
@@ -1,27 +1,83 @@
 import React from 'react';
+import classnames from 'classnames';
 import Button from '../../../ui/button';
+import CommandCopier from '../../../ui/command-copier/command-copier';
+import CutIcon from '../../../icons/cut';
+import IconButton from '../../../ui/icon-button';
+import { sidebarWidth, metaSidebarWidth } from '../../../../config';
 
 import './sliced-pipeline-action-bar.scss';
 
-export const SlicedPipelineActionBar = ({
-  chartSize,
-  slicedPipeline,
-  visibleSidebar,
-}) => {
-  const { outerWidth } = chartSize;
+export const SlicedPipelineActionBar = React.forwardRef((props, ref) => {
+  const {
+    chartSize,
+    displayMetadataPanel,
+    isSlicingPipelineApplied,
+    onApplySlicingPipeline,
+    onResetSlicingPipeline,
+    runCommand,
+    slicedPipeline,
+    visibleSidebar,
+  } = props;
+  const { outerWidth: screenWidth } = chartSize;
+  const buffer = 200;
+  const slicePipelineActionBarWidth =
+    ref.current?.firstChild.getBoundingClientRect().width;
+  const metaDataPanelWidth = displayMetadataPanel
+    ? metaSidebarWidth.open + buffer
+    : metaSidebarWidth.open;
+  const nodeListWidth = visibleSidebar
+    ? sidebarWidth.open - buffer
+    : sidebarWidth.open - buffer / 2;
+  const minimumTransformX = visibleSidebar
+    ? sidebarWidth.open
+    : sidebarWidth.closed;
+
+  // Calculate the transformX value ensuring it does not go below the minimum required for the sidebar opened and closed
+  const transformX = Math.max(
+    screenWidth -
+      nodeListWidth -
+      metaDataPanelWidth -
+      slicePipelineActionBarWidth / 2,
+    minimumTransformX
+  );
 
-  const transformX = visibleSidebar ? outerWidth / 2 + 100 : outerWidth / 2;
   return (
     <div
       className="sliced-pipeline-action-bar"
       style={{ transform: `translateX(${transformX}px)` }}
     >
       <div className="sliced-pipeline-action-bar--info">
-        {`${slicedPipeline.length} selected`}
+        {`${slicedPipeline.length} ${
+          isSlicingPipelineApplied ? 'sliced' : 'selected'
+        }`}
       </div>
-      <div className="sliced-pipeline-action-bar--cta">
-        <Button>Slice</Button>
+      <div
+        className={classnames('sliced-pipeline-action-bar--run-command', {
+          'sliced-pipeline-action-bar--run-command-long':
+            runCommand.length > 85,
```

Comment:
can we move 85 to config or constants file and also, how did we come up to this number ?

Reply:
it's just a magic number. from look and feel anything more than 85 seems like a long command, and no need to truncate. I'll give it a name so it makes more sense we read it